hack/local-up-cluster.sh

```diff
@@ -884,8 +884,18 @@ EOF
 function start_kubedns {
     if [[ "${ENABLE_CLUSTER_DNS}" = true ]]; then
         cp "${KUBE_ROOT}/cluster/addons/dns/kube-dns/kube-dns.yaml.in" kube-dns.yaml
-        sed -i -e "s/{{ pillar\['dns_domain'\] }}/${DNS_DOMAIN}/g" kube-dns.yaml
-        sed -i -e "s/{{ pillar\['dns_server'\] }}/${DNS_SERVER_IP}/g" kube-dns.yaml
+
+        # Fixes compatibility between BSD and GNU sed b/c this script is rather frequently used locally.
+        case "$(uname -s)" in
+          Darwin)
+            sed -i "" "s/{{ pillar\['dns_domain'\] }}/${DNS_DOMAIN}/g" kube-dns.yaml
```

Comment:
The GNU vs BSD sed strikes again. :)

But wow that this has lingered for so long in the local-up-cluster script! I'm surprised we didn't find this sooner.

Instead of checking this here, we can use `ensure-gnu-sed`:

https://github.com/kubernetes/kubernetes/blob/488f0fcda3f9baaab560435488ec9fd0a83401b1/hack/lib/util.sh#L782-L797

`ensure-gnu-sed` is being used in https://github.com/kubernetes/kubernetes/blob/master/test/images/image-util.sh#L52 nicely.

Reply:
> Instead of checking this here, we can use ensure-gnu-sed:

💯nice catch! much cleaner w/ this util. and ptal if i'm using correctly.